### PR TITLE
Support reading images from memory views

### PR DESCRIPTION
**Pull request by orf**
Currently reading an image from a buffer (`memoryview(some_bytes)`) fails. This PR aims to support reading memory views.

### CODE REVIEW

File: imageio/core/request.py [hunk 1]
```diff
@@ -183,14 +183,10 @@ def _parse_uri(self, uri):
                 elif len(uri) < 256:  # Can go wrong with veeery tiny images
                     self._uri_type = URI_FILENAME
                     self._filename = uri
-                elif isinstance(uri, binary_type) and is_read_request:
-                    self._uri_type = URI_BYTES
-                    self._filename = "<bytes>"
-                    self._bytes = uri
```
Comment (orf) [inline imageio/core/request.py]: This was a duplicate of the py3k branch below, so I removed it.

Comment (almarklein) [reply imageio/core/request.py]: I don't think it is, it's indented further. Also note that `binary_type` overlaps with `string_types` on Python 2. We can get rid of all this stuff somewhere this year, when we stop supporting Python 2 :)

**Comment by almarklein** [comment]
Nice addition!

**Comment by almarklein** [comment]
Done via #427. Thanks for this!

**Pull request closed without merging**